src/Avalonia.Base/Media/Fonts/UnmanagedFontMemory.cs

```diff
@@ -0,0 +1,407 @@
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Avalonia.Media.Fonts
+{
+    /// <summary>
+    /// Represents a memory manager for unmanaged font data, providing functionality to access and manage font memory
+    /// and OpenType table data.
+    /// </summary>
+    /// <remarks>This class encapsulates unmanaged memory containing font data and provides methods to
+    /// retrieve specific OpenType table data. It ensures thread-safe access to the memory and supports pinning for
+    /// interoperability scenarios. Instances of this class must be properly disposed to release unmanaged
+    /// resources.</remarks>
+    internal sealed unsafe class UnmanagedFontMemory : MemoryManager<byte>, IFontMemory
+    {
+        private IntPtr _ptr;
+        private int _length;
+        private bool _disposed;
```

Comment:
The `_disposed` flag can probably be removed: `_ptr == IntPtr.Zero` already does the same thing.